cis194/week4/tien/homework4/src/Homework4.hs

```diff
@@ -30,3 +30,33 @@ data Tree a
   = Leaf
   | Node Integer (Tree a) a (Tree a)
   deriving (Show, Eq)
+
+checkDepth :: Eq a => Tree a -> Integer
+checkDepth Leaf = (0 :: Integer)
+checkDepth (Node _ Leaf _ Leaf) = 1
+checkDepth (Node _ left _ right)
+  | left /= Leaf && right == Leaf = 1 + checkDepth left
+  | right /= Leaf && left == Leaf = 1 + checkDepth right
```

Comment:
Do we need `right /= Leaf` and `left /= Leaf` in these 2 guards?

Reply:
Refined the pattern matching.